device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
 ## Overview
 Object used to subscribe to the Cloud-to-Device messages for Twin
 
-## Example
-```javascript
-var receiver = new AmqpTwinClient(config: ClientConfig, client: any);
-receiver.on('response', function(response) {
-    console.log('Response received for request ' + response.requestId);
-    console.log('  status = " + response.status);
-    console.log(response.body);
-});
+## Public API
+
+```typescript
+class AmqpTwinClient extends EventEmitter {
+  constructor(authenticationProvider: AuthenticationProvider, client: any) {
+  getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
+  updateTwinReportedProperties(patch: any, callback: (err?: Error) => void): void;
+  enableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  disableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  detach(callback: (err?: Error) => void): void;
+}
 ```
+### constructor(authenticationProvider: AuthenticationProvider, client: any) {
 
-## Public API
+**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
```

Comment:
>SRS_NODE_DEVICE_AMQP_TWIN_06_005 [](start = 2, length = 32)

OK, but this seems out of scope for what we usually specify.  This seems like an implementation detail or a comment about how the code is structured, not something about how the code behaves.

Reply:
yes and no, it's also a good way to carry the idea that this thing is going to fire events I think we have it in other places in the code too for other event emitters.

---
In reply to: [171681955](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171681955) [](ancestors = 171681955)